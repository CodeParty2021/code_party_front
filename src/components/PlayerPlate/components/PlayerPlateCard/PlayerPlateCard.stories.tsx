import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { PlayerPlateCard } from "./PlayerPlateCard";

export default {
  title: "components/PlayerPlate/PlayerPlateCard",
  component: PlayerPlateCard,
} as ComponentMeta<typeof PlayerPlateCard>;

const Template: ComponentStory<typeof PlayerPlateCard> = (args) => (
  <PlayerPlateCard {...args} />
);

export const Default = Template.bind({});
Default.args = {
  playerIcon: <>icon</>,
  viewAlgo: true,
  columnTop: <>top</>,
  columnMiddle: <>middle</>,
  columnBottom: <>bottom</>,
};
