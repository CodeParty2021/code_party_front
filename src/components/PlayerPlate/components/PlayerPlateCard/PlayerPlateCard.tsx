import { Algo } from "components/Character/Algo/Algo";
import React, { ReactElement } from "react";
import {
  PlayerPlateCardStyle,
  PlayerPlateCardStyleProps,
} from "./PlayerPlateCardStyle";

type Props = PlayerPlateCardStyleProps & {
  playerIcon: ReactElement;
  viewAlgo: boolean;
  columnTop: ReactElement;
  columnMiddle?: ReactElement;
  columnBottom: ReactElement;
};

export const PlayerPlateCard: React.FC<Props> = ({
  playerIcon,
  viewAlgo,
  columnTop,
  columnMiddle,
  columnBottom,
  ...styleProps
}) => {
  return (
    <PlayerPlateCardStyle {...styleProps}>
      <div className="playerplatecard_window">
        {viewAlgo && <Algo color={styleProps.color} />}
      </div>
      <div className="playerplatecard_column_top">{columnTop}</div>
      {columnMiddle && (
        <div className="playerplatecard_column_middle">{columnMiddle}</div>
      )}
      <div className="playerplatecard_column_bottom">{columnBottom}</div>

      {/* absolute要素 */}
      <div className="playerplatecard_playericon">{playerIcon}</div>
      <div className="playerplatecard_border" />
    </PlayerPlateCardStyle>
  );
};
