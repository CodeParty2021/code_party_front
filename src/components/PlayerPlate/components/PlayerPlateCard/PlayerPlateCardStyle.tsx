import { AlgoStyle } from "components/Character/Algo/AlgoStyle";
import { PlayerPlateColors } from "components/PlayerPlate/colors";
import styled, { css } from "styled-components";
import { FlexGap } from "styles/FlexGap/FlexGap";

export type PlayerPlateCardStyleProps = {
  color?: "turquoise" | "leaf" | "orange" | "magenta";
  status?: "default" | "ready" | "disconnecting" | "bot" | "waiting";
  size?: "L" | "M";
};

const defaultStyle = css`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 32px;
  ${FlexGap({ gap: "16px", direction: "column" })}

  position: relative;

  border-radius: 48px;
  background-color: ${PlayerPlateColors.background};

  .playerplatecard_playericon {
    position: absolute;
    top: 30px;
    left: 30px;
  }

  .playerplatecard_window {
    position: relative;
    border-radius: 24px;
    align-self: stretch;
    flex-grow: 1;
    overflow: hidden;

    ${AlgoStyle} {
      position: absolute;
      bottom: -48px;
      left: 50%;
      transform: translateX(-50%);
    }
  }

  .playerplatecard_column_top,
  .playerplatecard_column_middle,
  .playerplatecard_column_bottom {
    width: 100%;
  }

  .playerplatecard_column_top,
  .playerplatecard_column_middle {
    border-bottom: 1px solid ${PlayerPlateColors.border};
  }

  .playerplatecard_column_bottom {
  }

  .playerplatecard_border {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;

    pointer-events: none;

    border: 1px solid ${PlayerPlateColors.border};
    border-radius: 48px;
  }
`;

const turquoiseStyle = css`
  .playerplatecard_window {
    background-color: ${PlayerPlateColors.turquoise.back};
  }

  .playerplatecard_border {
    border-color: ${PlayerPlateColors.turquoise.main};
  }
`;

const leafStyle = css`
  .playerplatecard_window {
    background-color: ${PlayerPlateColors.leaf.back};
  }

  .playerplatecard_border {
    border-color: ${PlayerPlateColors.leaf.main};
  }
`;

const orangeStyle = css`
  .playerplatecard_window {
    background-color: ${PlayerPlateColors.orange.back};
  }

  .playerplatecard_border {
    border-color: ${PlayerPlateColors.orange.main};
  }
`;

const magentaStyle = css`
  .playerplatecard_window {
    background-color: ${PlayerPlateColors.magenta.back};
  }

  .playerplatecard_border {
    border-color: ${PlayerPlateColors.magenta.main};
  }
`;

const defaultBorderStyle = css`
  .playerplatecard_border {
    border-color: ${PlayerPlateColors.border};
  }
`;

const readyStyle = css`
  .playerplatecard_border {
    border-width: 4px;
  }
`;

const disconnectingStyle = css`
  .playerplatecard_window {
    background-color: ${PlayerPlateColors.gray.back};
  }

  .playerplatecard_border {
    border-color: ${PlayerPlateColors.border};
  }
`;

const botStyle = css`
  .playerplatecard_window {
    background-color: ${PlayerPlateColors.gray.back};
  }

  .playerplatecard_border {
    border-width: 4px;
  }
`;

const waitingStyle = css`
  .playerplatecard_window {
    background-color: ${PlayerPlateColors.gray.back};
  }

  .playerplatecard_border {
    border-color: ${PlayerPlateColors.border};
  }
`;

const lStyle = css`
  width: 432px;
  height: 588px;
`;

const mStyle = css`
  width: 320px;
  height: 408px;
`;

export const PlayerPlateCardStyle = styled.div<PlayerPlateCardStyleProps>`
  ${defaultStyle}

  ${({ color }) => color === "turquoise" && turquoiseStyle}
  ${({ color }) => color === "leaf" && leafStyle}
  ${({ color }) => color === "orange" && orangeStyle}
  ${({ color }) => color === "magenta" && magentaStyle}

  ${({ status }) => status === "default" && defaultBorderStyle}
  ${({ status }) => status === "ready" && readyStyle}
  ${({ status }) => status === "disconnecting" && disconnectingStyle}
  ${({ status }) => status === "bot" && botStyle}
  ${({ status }) => status === "waiting" && waitingStyle}

  ${({ size }) => size === "L" && lStyle}
  ${({ size }) => size === "M" && mStyle}
`;

PlayerPlateCardStyle.defaultProps = {
  color: "turquoise",
  status: "default",
  size: "M",
};
