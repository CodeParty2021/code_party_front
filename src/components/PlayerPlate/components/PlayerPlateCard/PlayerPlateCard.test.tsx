import React from "react";
import { PlayerPlateCard } from "./PlayerPlateCard";
import "jest-styled-components";
import { render } from "enzyme";

describe("<PlayerPlateCard />", () => {
  it("snapshot test", () => {
    const wrapper = render(
      <PlayerPlateCard
        playerIcon={<>icon</>}
        viewAlgo={true}
        columnTop={<>top</>}
        columnMiddle={<>middle</>}
        columnBottom={<>bottom</>}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
