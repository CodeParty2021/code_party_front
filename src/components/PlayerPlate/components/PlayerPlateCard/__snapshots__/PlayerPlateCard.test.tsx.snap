// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerPlateCard /> snapshot test 1`] = `
.c2 .algo_color_main {
  fill: #34C5D5;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 32px;
  gap: 16px;
  position: relative;
  border-radius: 48px;
  background-color: #FAFAFA;
  width: 320px;
  height: 408px;
}

.c0 .playerplatecard_playericon {
  position: absolute;
  top: 30px;
  left: 30px;
}

.c0 .playerplatecard_window {
  position: relative;
  border-radius: 24px;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: hidden;
}

.c0 .playerplatecard_window .c1 {
  position: absolute;
  bottom: -48px;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c0 .playerplatecard_column_top,
.c0 .playerplatecard_column_middle,
.c0 .playerplatecard_column_bottom {
  width: 100%;
}

.c0 .playerplatecard_column_top,
.c0 .playerplatecard_column_middle {
  border-bottom: 1px solid #D7DAE4;
}

.c0 .playerplatecard_border {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  pointer-events: none;
  border: 1px solid #D7DAE4;
  border-radius: 48px;
}

.c0 .playerplatecard_window {
  background-color: #A7ECF1;
}

.c0 .playerplatecard_border {
  border-color: #34C5D5;
}

.c0 .playerplatecard_border {
  border-color: #D7DAE4;
}

@supports not (gap:16px) {
  .c0 > *:not(:first-child) {
    margin-top: 16px;
  }
}

<div
  class="c0"
  color="turquoise"
>
  <div
    class="playerplatecard_window"
  >
    <div
      class="c1 c2"
      color="turquoise"
    />
  </div>
  <div
    class="playerplatecard_column_top"
  >
    top
  </div>
  <div
    class="playerplatecard_column_middle"
  >
    middle
  </div>
  <div
    class="playerplatecard_column_bottom"
  >
    bottom
  </div>
  <div
    class="playerplatecard_playericon"
  >
    icon
  </div>
  <div
    class="playerplatecard_border"
  />
</div>
`;
